build.gradle

```diff
@@ -28,6 +28,7 @@ ext {
             test: [
                     'commons-io:commons-io:2.11.0',
                     'junit:junit:4.13.2',
+                    'pl.pragmatists:JUnitParams:1.1.1',
```

Comment:
Please don't add new dependencies.

Reply:
@kallestenflo , This dependency provides the `JUnitParamsRunner` used in `JacksonJsonNodeJsonProviderMapperSupportTest`. Without this test runner I'll have to divide the test case in three different classes, as JUnit 4's default `Parameterized` test runner can only pass parameters at class level and not at test method level.

Moreover since this a test dependency only it shouldn't build size for any client of this lib. Let me know if we still want me remove this dependency.

Reply:
@kallestenflo , Any update on above?

Reply:
@kallestenflo , Any update on above?

Reply:
@kallestenflo , Any update on above?

Reply:
@debanshuk I don't want new dependencies added to the project. Once a dependency is added it's hard to drop. Please solve in an other way.

Reply:
@kallestenflo , I have remove the dependency and have update the test case. I was able to work around the constraint of  class level test data (without any method level support) in JUnit 4's parameterized test, by cloning `DocumentContext` object before executing each test method. This prevented modification of input `DocumentContext` by a test method, and allowed me to use same test data in multiple test methods. Kindly review the updated commit.

Reply:
@kallestenflo , Any update on above?

Reply:
@kallestenflo , Any update on above?

Reply:
@kallestenflo , Any update on above?

Reply:
@kallestenflo , Any update on above?

Reply:
@kallestenflo , Any update on above? I have done the changes as per your comment. Kindly approve the CR, if there aren't any further comments.